### PR TITLE
on-device decisioning telemetry

### PR DESCRIPTION
**Pull request by jasonwaters**
https://wiki.corp.adobe.com/pages/viewpage.action?spaceKey=elm&title=Local+Decisioning%3A+SDK+telemetry

- added telemetry for on-device decisioning
- changed default artifact format to JSON

Example of telemetry sent...
screenshot-2020-09-16-115209@2x

### CODE REVIEW

**Comment by artur-ciocanu** [comment]
@jasonwaters in the WIKI related to telemetry in Phase 1, we have just:
- request ID
- timestamp
- execution
- fetaures - with just decisioningMethod

I would keep it this way, until we have clarity around Phase 2 etc.

**Comment by jasonwaters** [comment]
> @jasonwaters in the WIKI related to telemetry in Phase 1, we have just:
> 
> * request ID
> * timestamp
> * execution
> * fetaures - with just decisioningMethod
> 
> I would keep it this way, until we have clarity around Phase 2 etc.

Ok, thanks for reviewing @artur-ciocanu .  I changed the flag to be `telemetryEnabled` and limited what is sent to just what is required for phase 1. 🍻

File: packages/target-decisioning-engine/types/DecisioningConfig.d.ts [hunk 1]
```diff
@@ -66,6 +66,11 @@ export interface DecisioningConfig {
    */
   sendNotificationFunc?: Function;
 
+  /**
+   * If set to true, telemetry data will not be sent to Adobe
+   */
```
Comment (artur-ciocanu) [inline packages/target-decisioning-engine/types/DecisioningConfig.d.ts:71]: Should this be renamed to `telemetryEnabled`

File: packages/target-decisioning-engine/types/TelemetryEntry.d.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+export interface TelemetryRequest {
```
Comment (artur-ciocanu) [inline packages/target-decisioning-engine/types/TelemetryEntry.d.ts]: As discussed, probably it's too early for these changes, let's remove them until we  have a chance to talk about Phase 2.

File: packages/target-nodejs-sdk/src/validators.js [hunk 3]
```diff
@@ -93,10 +93,16 @@ export function validateSendNotificationsOptions(options) {
...
 
-  const { notifications } = request;
+  const { notifications, telemetry } = request;
 
   if (isEmptyArray(notifications)) {
-    return Messages.NOTIFICATIONS_REQUIRED;
+    if (isUndefined(telemetry)) {
+      return Messages.NOTIFICATIONS_REQUIRED;
+    }
+
```
Comment (artur-ciocanu) [inline packages/target-nodejs-sdk/src/validators.js]: Telemetry is internal to our SDK, I don't think we need to add validation.

**Review by artur-ciocanu — CHANGES_REQUESTED** [review]
@jasonwaters it looks great, my only concern is:
- validation  for telemetry, I don't think we need it
- type definitions for timing and other stuff, that we will probably have in Phase 2, not sure we need to publish those at this time.

**Pull request merged**